client/allocrunner/taskrunner/template/template.go

```diff
@@ -194,7 +209,6 @@ func NewTaskTemplateManager(config *TaskTemplateManagerConfig) (*TaskTemplateMan
 	tm.runner = runner
 	tm.lookup = lookup
 
-	go tm.run()
```

Comment:
I know we had to do this because of the slight rearchitecture of the template hook, but I really like that we're not spawning a goroutine in the constructor anymore. It makes testing and reasoning about order-of-operations a lot easier. :+1: